### PR TITLE
Make it compile on FreeBSD.

### PR DESCRIPTION
**Pull request by fbettag**
I've taken the liberty to make this compile on FreeBSD (10.2-RELEASE tested).

### CODE REVIEW

**Comment by Kaushalrocks** [comment]
Can u tell me what u r speaking about??
Is there any update for pixiewps pc version for cracking dlink models like
600 m?? Plz reply asap

**Comment by wiire** [comment]
Thank you for contributing, however these changes break compiling on other evironments. If all it needs for compiling on FreeBSD are libraries to address endianness issues, then it can be simply achieved by using a simple runtime check like the one I wrote for Android (https://github.com/wiire/pixiewps/commit/d917f8f940c9ca1bd297ad6400d7841469049af4). I left the old preprocessor checks with the various libraries mainly because of my inexperience with endianness related issues in general but it seems to work fine when I tested.

I'm going to push some changes. Let me know if work for you.

File: src/random_r.c [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 #ifdef __MACH__
 # include "../include/features.h"
+#elif __unix__
```
Comment (wiire) [inline src/random_r.c]: What does this line accomplish?

Comment (fbettag) [inline src/random_r.c]: include literally nothing on unix :) apparently there's everything already there that's needed.

Comment (fbettag) [inline src/random_r.c]: Well i could go deeper into checking for FreeBSD and NetBSD and OpenBSD, true. But i'm pretty sure they all run the same way. __LINUX__ should not classify as __unix__ tho :s

**Comment by wiire** [comment]
I pushed the changes (https://github.com/wiire/pixiewps/commit/6f42dc1279da3a9c32fc05338dfecdfe5a4dc3b4). Could you try to compile the program again with these new changes? Thank you.

**Comment by fbettag** [comment]
Nope, says "features.h not found". There is none on BSD. :)

**Comment by fbettag** [comment]
maybe make it elif MACH to get rid of the empty #elif unix

**Comment by wiire** [comment]
Yes, sorry I didn't know what that directive was there for. To keep the `features.h` present, even if it is not strictly necessary, I see two alternatives:

include it, like it was done for Mac OS:
```
#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
# include <sys/param.h>
# if defined(BSD) || (defined(__APPLE__) && defined(__MACH__))
#  include "../include/features.h"
# else
#  include <features.h>
# endif
#endif
```

or get rid of it on *BSD OS:
```
#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
# include <sys/param.h>
# if defined(__APPLE__) && defined(__MACH__)
#  include "../include/features.h"
# elif(BSD)
/* BSD (DragonFly BSD, FreeBSD, OpenBSD, NetBSD)
   Nothing to include */
# else
#  include <features.h>
# endif
#endif
```

I used this as a reference: http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system

I assume you use __Clang/LLVM__ as opposed to __GCC__? Anyway let me know what do you think it's best or if you have any suggestion.

**Comment by fbettag** [comment]
Yup, Clang is right.

Variant 1 doesnt look good to me, i'd rather go with 2.

**Comment by Kaushalrocks** [comment]
But what is the use for running pixie on free bsd ? Is there any advantage??

**Comment by fbettag** [comment]
What's the use for running it on Linux? Why would i want it on BSD? Take your best guess.

**Comment by wiire** [comment]
Done (https://github.com/wiire/pixiewps/commit/08a78c3c2712523d9106c3a20e31c081dabe558a). Thank you again for contributing.

**Comment by Kaushalrocks** [comment]
Please wiire can u answer is freebsd is better than kali 2.0?

**Pull request closed without merging**